src/components/app/userActionViewKeyPageDeeplinkWrapper/loading.tsx

```diff
@@ -0,0 +1,74 @@
+'use client'
+
+import { useEffect, useState } from 'react'
+import { useRouter } from 'next/navigation'
+
+import { actionCreateUserActionViewKeyPage } from '@/actions/actionCreateUserActionViewKeyPage'
+import { LoadingOverlay } from '@/components/ui/loadingOverlay'
+import { SupportedCountryCodes } from '@/utils/shared/supportedCountries'
+
+type Status = 'waiting' | 'ready'
+
+export interface CampaignMetadata {
+  url: string
+}
+
+interface UserActionViewKeyPageDeeplinkLoadingProps {
+  campaignMetadata: CampaignMetadata
+  campaignName: string
+  countryCode: SupportedCountryCodes
+  minWaitTimeInSeconds?: number
+  pathName: string
+}
+
+const MIN_WAIT_TIME_IN_SECONDS = 1
+
+export function UserActionViewKeyPageDeeplinkLoading({
+  campaignMetadata,
+  campaignName,
+  countryCode,
+  minWaitTimeInSeconds = MIN_WAIT_TIME_IN_SECONDS,
+  pathName,
+}: UserActionViewKeyPageDeeplinkLoadingProps) {
+  const router = useRouter()
+
+  const [status, setStatus] = useState<Status>('waiting')
+
+  const [actionHasBeenCreated, setActionHasBeenCreated] = useState(false)
+
+  useEffect(() => {
+    const minWaitTimeTimeout = setTimeout(() => {
+      setStatus('ready')
+    }, minWaitTimeInSeconds * 1_000)
+
+    return () => {
+      clearInterval(minWaitTimeTimeout)
+    }
+  }, [minWaitTimeInSeconds])
```

Comment:
np: I'd suggest separating this part into a custom hook, an api like this below would look way better to me
```suggestion
  useTimeout(() => setStatus('ready'), minWaitTimeInSeconds * 1000)
  // ...
  
  // bottom of the file
  function useTimeout(callback, ms) {
    useEffect(() => {
      const timeoutRef = setTimeout(callback, ms)

      return () => {
        clearTimeout(timeoutRef)
      }
    }, [callback, ms])
  }
```